src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -98,6 +104,14 @@ protected Object readResolve() throws IOException {
         return this;
     }
 
+    /** Allows user to disable saving step info, if this imposes too high a performance overhead
+     *   by setting System Property org.jenkinsci.plugins.workflow.cps.DSL.keepStepInfo to 'false'.
+     *  This may be modified at runtime by the user if needed for debugging. */
+    private static boolean isKeepStepInfo() {
+        String prop = System.getProperty(DSL.class.getName()+".keepStepInfo");
```

Comment:
Bad pattern for a run-time property. Normal pattern is a static non-final field initialized to the system property. Jenkins admins will be confused.